orttraining/orttraining/python/training/ortmodule/_io.py

```diff
@@ -129,8 +129,20 @@ def _combine_input_buffers_initializers(params, onnx_input_names, input_info, bu
         * Initializers: computed from original PyTorch model parameters
     '''
 
+    def _expand_inputs(current_input, non_none_inputs):
+        if isinstance(current_input, abc.Sequence):
+            # If the input is a sequence (like a list), expand the list so that
+            # each element of the list is an input by itself
+            for inp in current_input:
+                _expand_inputs(inp, non_none_inputs)
+        elif current_input is not None:
+            # else just collect all the non none inputs within non_none_inputs
+            non_none_inputs.append(current_input)
+
+
     # User inputs
-    non_none_inputs = [inp for inp in inputs if inp is not None]
+    non_none_inputs = []
+    _expand_inputs(inputs, non_none_inputs)
```

Comment:
What would it take to support list-valued arguments in `kwargs` as well?

Reply:
Let me try to add support for ```kwargs``` argument being a list.

Reply:
Support for ```kwargs``` might need more work than I had anticipated. I'll create another work item to address support for lists in ```kwargs```. For now, this PR unblocks some of the models that need lists as input to ```ORTModule```, so we could decouple the two.